.github/workflows/publish-to-pypi.yml

```
@@ -2,15 +2,11 @@ name: Publish to PyPI
 
 on:
   release:
-    types: [released]
+    types: [ released ]
```

Why limit this to just releasing? Really this should be run (without uploading) on every PR and be manually triggerable.
```suggestion
    types: [released, pull_request, workflow_dispatch]
```